src/app/helpers/encoding/encoding.ts

```diff
@@ -1,28 +1,43 @@
 /**
- * TODO Finish comment (Base64 RFC 4648 §4???)
- * Encode plain text into base64url following the TODO standard
+ * Encode plain text into base64url following the RFC 4648 §5 (base64url)
+ * standard. Returns empty string during SSR
  *
  * @param plainText Plain text to encode
  */
 export function toBase64Url(plainText: string): string {
-  // Convert to base64, then replace unsafe symbols
+  if (!window?.btoa) {
+    return "";
+  }
+
+  // Encode special characters (btoa only supports Latin1)
+  //  then convert to base64 (RFC 4648 §4),
+  //  then replace so we conform to the base64url spec (RFC 4648 §5)
   return window
-    .btoa(plainText)
+    .btoa(encodeURIComponent(plainText))
```

Comment:
HMMM this is not ok.

These won't be decoded server side.

The reason we didn't have to do this was because JSON.stringify handles encoding outside of our supported range. So using both stringify and btoa safely side steps the problem.

As far as I know btoa treats it's input as a series of bytes; there should be no case where we have to worry about this. Can you explain/show me why you made this change?

Reply:
As stated in the comment above, `btoa` only supports Latin1. While `JSON.stringify` handles encoding for us, it is not directly used in this function so cannot be taken as an assumption. We could make it take in an object as an input instead of text? Also, if `JSON.stringify` encodes the text, why will that be decoded server side, but not `encodeURIComponent`?